lib/datadog/tracing/contrib/sinatra/patcher.rb

```diff
@@ -58,7 +58,7 @@ def patch
           end
 
           def register_tracer
-            ::Sinatra.send(:register, Contrib::Sinatra::Tracer)
+            ::Sinatra::Base.register(Contrib::Sinatra::Tracer)
             ::Sinatra::Base.prepend(Sinatra::Tracer::Base)
```

Comment:
This is an interesting change -- do you know why we didn't do it like this before?

Reply:
I think the implementation was only partially working for classic Sinatra, but not those modular ones. Also the reason we ask user to `register Datadog::Tracing::Contrib::Sinatra::Tracer` explicitly. Otherwise, the instrumentation would be missing.